xdsl/dialects/arm_cf.py

```diff
@@ -0,0 +1,66 @@
+from xdsl.dialects import arm
+from xdsl.dialects.arm.assembly import AssemblyInstructionArg
+from xdsl.dialects.arm.ops import LabelOp
+from xdsl.dialects.arm.register import IntRegisterType
+from xdsl.dialects.builtin import StringAttr
+from xdsl.ir import Dialect, Operation, SSAValue
+from xdsl.irdl import (
+    Successor,
+    irdl_op_definition,
+    operand_def,
+    successor_def,
+    traits_def,
+)
+from xdsl.traits import IsTerminator
+
+
+@irdl_op_definition
+class BEqOp(arm.ops.ARMInstruction):
+    """
+    Branch if equal.
+    """
+
+    name = "arm_cf.beq"
+
+    s1 = operand_def(IntRegisterType)
+    s2 = operand_def(IntRegisterType)
+
+    then_block = successor_def()
+
+    assembly_format = (
```

Comment:
I'm aware that this is not correct, but I'm not sure how exactly to fix it. In arm, `beq` must follow a `cmp` in the line beforehand. (Related question: Shall I still treat is as one instruction and just include the `cmp` in the printing, or implement cmp separately?)

Reply:
I think I'd add the `cmp` separately and have a verification check here in `beq` the enforces this constraint (i.e., the `cmp` must be the instruction prior to this), similarly to some of the `verify_` methods in the RISC-V dialect.
We don't have many there and the constraints are a bit more involved, but you should get an idea.

Let me know if you have more questions on this.

Reply:
I'm not 100% sure, if the constraint is so hard-coded, then maybe we could get away with a single op. Probably worth a discussion.

Reply:
Well, it depends on the CPSR AFAIU, and I'm not sure how to model that fully, e.g., when these are cleared, as the specification phrases this as "set by the last flag-setting instruction... etc.". For example, the `cmp` instruction sets the Z flag.